docs/topics/i18n/translation.txt

```diff
@@ -1791,9 +1791,18 @@ redirect to that URL will be performed. Otherwise, Django may fall back to
 redirecting the user to the URL from the ``Referer`` header or, if it is not
 set, to ``/``, depending on the nature of the request:
 
-* For AJAX requests, the fallback will be performed only if the ``next``
-  parameter was set. Otherwise a 204 status code (No Content) will be returned.
-* For non-AJAX requests, the fallback will always be performed.
+* If the request accepts HTML content (based on its ``Accept`` header), the
+  fallback will always be performed.
+* If the request doesn't accept HTML, the fallback will be performed only if
+  the ``next`` parameter was set. Otherwise a 204 status code (No Content) will
+  be returned.
+
+.. versionchanged:: 3.1
+
+    In older versions, the distinction for the fallback was based on whether
+    the ``X-Requested-With`` header was set to the value ``XMLHttpRequest``.
+    This is set by the jQuery ``ajax()`` method.
```

Comment:
Since this is (basically) backwards incompatible, should we have a release note too?